lib/slack-ruby-bot/app.rb

```diff
@@ -3,26 +3,36 @@
 module SlackRubyBot
   class App < Server
     def initialize(options = {})
-      SlackRubyBot.configure do |config|
-        config.token = ENV['SLACK_API_TOKEN'] || raise("Missing ENV['SLACK_API_TOKEN'].")
-        config.aliases = ENV['SLACK_RUBY_BOT_ALIASES'].split(' ') if ENV['SLACK_RUBY_BOT_ALIASES']
-      end
-      Slack.configure do |config|
-        config.token = SlackRubyBot.config.token
-      end
-      super
+      server_options = { token: options[:token] }
+      super(server_options)
     end
 
     def config
       SlackRubyBot.config
     end
 
     def self.instance
-      @instance ||= new
+      @instance ||= begin
+        configure!
+        new(token: SlackRubyBot.config.token)
+      end
     end
 
     private
 
+    def self.configure!
+      SlackRubyBot.configure do |config|
+        config.token = ENV['SLACK_API_TOKEN'] if ENV.key?('SLACK_API_TOKEN')
+        raise('Missing Slack API Token.') if config.token.nil?
+
+        config.aliases = ENV['SLACK_RUBY_BOT_ALIASES'].split(' ') if ENV['SLACK_RUBY_BOT_ALIASES']
+      end
+      Slack.configure do |config|
+        config.token = SlackRubyBot.config.token
+      end
+    end
+    private_class_method :configure!
```

Comment:
Let's keep it public so users can call it without using a singleton instance? No need to document.